src/messages.rs

```diff
@@ -187,12 +187,30 @@ impl HopMessage {
 }
 
 /// A list of a group's public IDs, together with a list of signatures of a neighbouring group.
-#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, RustcEncodable, RustcDecodable)]
+#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, RustcEncodable, RustcDecodable, Debug)]
 pub struct GroupList {
     // TODO(MAID-1677): pub signatures: BTreeSet<(PublicId, sign::Signature)>,
     pub pub_ids: BTreeSet<PublicId>,
 }
 
+impl GroupList {
+    /// Returns true if our name is the `route`-th closest to `src_name` in our group.
+    ///
+    /// Used when sending a message from a group to decide which one of the group should send the
+    /// full message (the remainder sending just a hash of the message).
+    pub fn should_route_full_message(&self,
+                                     our_name: &XorName,
+                                     dst_name: &XorName,
+                                     route: usize)
+                                     -> bool {
+        let our_group = self.pub_ids
+            .iter()
+            .map(|id| id.name())
+            .sorted_by(|&lhs, &rhs| dst_name.cmp_distance(lhs, rhs));
+        *our_group[route % our_group.len()] == *our_name
```

Comment:
We need to make sure to also use the group list for determining where to send the signature to. Probably `RoutingTable::get_routeth_node` should also move into `GroupList`?